pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -0,0 +1,645 @@
+package sqlstore
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"gorm.io/gorm"
+)
+
+const (
+	// MySQL database type
+	MySQL = "mysql"
+	// PostgreSQL database type
+	PostgreSQL = "postgres"
+	// SQLite database type
+	SQLite = "sqlite3"
+)
+
+func OpenDB(connectionString, dbtype string) (db *gorm.DB, err error) {
+	var dialectvar dialect
+
+	switch dbtype {
+	case SQLite:
+		dialectvar = sqliteDB{}
+	case PostgreSQL:
+		dialectvar = postgresDB{}
+	case MySQL:
+		dialectvar = mysqlDB{}
+	default:
+		return nil, fmt.Errorf("unsupported database_type: %s" + dbtype)
+	}
+	db, err = dialectvar.connect(connectionString)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to: %s", connectionString)
+	}
+	return db, nil
+}
+
+func CreateAllTablesinDB(db *gorm.DB) (err error) {
+	err = CreateOrganizationTableInDB(db)
+	if err != nil {
+		return err
+	}
```

Comment:
nit: we could simplify all err handlings in this file
```suggestion
	if err := CreateOrganizationTableInDB(db); err != nil;  {
		return err
	}
```